src/main/scala/org/broadinstitute/clio/dataaccess/CachedServerStatusDAO.scala

```diff
@@ -0,0 +1,47 @@
+package org.broadinstitute.clio.dataaccess
+
+import org.broadinstitute.clio.model.ServerStatusInfo
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+  * Stores the server status in a synchronized mutable variable.
+  */
+class CachedServerStatusDAO private() extends ServerStatusDAO {
```

Comment:
Hi - I haven't fully looked at this PR (or even know what this repo is for :)), but my [Rawls PR](https://github.com/broadinstitute/rawls/pull/703) is more about asynchronous monitoring of subsystems to provide health checks. It seems like this is more about lifecycle management? So maybe slightly different use cases, though I'm happy to discuss the possibility of sharing code.

As an aside, I've used [Guava service](https://github.com/google/guava/wiki/ServiceExplained) before as a way to manage services which have a lifecycle/state and found it pretty nice to use (granted this was in a Java project).